src/test/java/seedu/address/logic/wanderlustlogic/wanderlustparser/WanderLustParserTest.java

```diff
@@ -0,0 +1,106 @@
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.ClearCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.ExitCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.FindCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.HelpCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.ListCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.add.AddActivityCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.delete.DeleteActivityCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.delete.DeleteCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.edit.EditActivityCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.edit.EditCommand;
+import seedu.address.logic.wanderlustlogic.wanderlustcommands.edit.EditDescriptor;
+import seedu.address.logic.wanderlustlogic.wanderlustparser.exceptions.ParseException;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.commons.NameContainsKeywordsPredicate;
+import seedu.address.testutil.ActivityUtil;
+import seedu.address.testutil.EditActivityDescriptorBuilder;
+import seedu.address.testutil.builders.ActivityBuilder;
+
+
+public class WanderLustParserTest {
+
+    private final WanderlustParser parser = new WanderlustParser();
+
+    @Test
+    public void parseCommand_add() throws Exception {
+        Activity activity = new ActivityBuilder().build();
+        AddActivityCommand command = (AddActivityCommand) parser.parseCommand(ActivityUtil.getAddCommand(activity));
+        Assertions.assertTrue(new AddActivityCommand(activity).equals(command));
+    }
+
+    @Test
+    public void parseCommand_clear() throws Exception {
+        assertTrue(parser.parseCommand(ClearCommand.COMMAND_WORD) instanceof ClearCommand);
+        assertTrue(parser.parseCommand(ClearCommand.COMMAND_WORD + " 3") instanceof ClearCommand);
+    }
+
+    @Test
+    public void parseCommand_delete() throws Exception {
+        DeleteActivityCommand command = (DeleteActivityCommand) parser.parseCommand(
+                DeleteCommand.COMMAND_WORD + " -activity " + INDEX_FIRST_TRAVELPLAN.getOneBased());
+        assertEquals(new DeleteActivityCommand(INDEX_FIRST_TRAVELPLAN), command);
+    }
+
+    @Test
+    public void parseCommand_edit() throws Exception {
+        Activity activity = new ActivityBuilder().build();
+        EditDescriptor descriptor = new EditActivityDescriptorBuilder(activity);
+        EditActivityCommand command = (EditActivityCommand) parser.parseCommand(EditCommand.COMMAND_WORD
+                + " -activity " + INDEX_FIRST_TRAVELPLAN.getOneBased() + " "
+                + ActivityUtil.getEditActivityDescriptorDetails(descriptor));
+        assertEquals(new EditActivityCommand(INDEX_FIRST_TRAVELPLAN, descriptor), command);
```

Comment:
I think it is throwing an exception because the edit command ensures that at least one field is edited but the command and builder is building the exact same fields so there wasn't any editing done

Reply:
Ok resolved. Thanks!